feature/gnmi/ate_tests/telemetry_port_speed_test/telemetry_port_speed_test.go

```diff
@@ -0,0 +1,371 @@
+		e.PortSpeed = portSpeed[p.Speed()]
+	}
+}
+
+func (tc *testCase) setupAggregateAtomically(t *testing.T) {
+	d := &telemetry.Device{}
+
+	if tc.lagType == lagTypeLACP {
+		d.GetOrCreateLacp().GetOrCreateInterface(tc.aggID)
+	}
+
+	agg := d.GetOrCreateInterface(tc.aggID)
+	agg.GetOrCreateAggregation().LagType = tc.lagType
+
+	for _, port := range tc.dutPorts {
+		i := d.GetOrCreateInterface(port.Name())
+		i.GetOrCreateEthernet().AggregateId = ygot.String(tc.aggID)
+	}
+
+	p := tc.dut.Config()
+	fptest.LogYgot(t, fmt.Sprintf("%s to Update()", tc.dut), p, d)
+	p.Update(t, d)
+}
+
+func (tc *testCase) clearAggregateMembers(t *testing.T) {
+	for _, port := range tc.dutPorts {
+		tc.dut.Config().Interface(port.Name()).Ethernet().AggregateId().Delete(t)
+	}
+}
+
+// sortPorts sorts the ports by the testbed port ID.
+func sortPorts(ports []*ondatra.Port) []*ondatra.Port {
+	sort.SliceStable(ports, func(i, j int) bool {
+		return ports[i].ID() < ports[j].ID()
+	})
+	return ports
+}
+
+func (tc *testCase) configureDUT(t *testing.T) {
+	t.Logf("dut ports = %v", tc.dutPorts)
+	if len(tc.dutPorts) < 2 {
+		t.Fatalf("Testbed requires at least 2 ports, got %d", len(tc.dutPorts))
+	}
+
+	d := tc.dut.Config()
+
+	if *deviations.AggregateAtomicUpdate {
+		tc.clearAggregateMembers(t)
+		tc.setupAggregateAtomically(t)
+	}
+
+	if tc.lagType == lagTypeLACP {
+		lacp := &telemetry.Lacp_Interface{Name: ygot.String(tc.aggID)}
+		lacp.LacpMode = telemetry.Lacp_LacpActivityType_ACTIVE
+
+		lacpPath := d.Lacp().Interface(tc.aggID)
+		fptest.LogYgot(t, "LACP", lacpPath, lacp)
+		lacpPath.Replace(t, lacp)
+	}
+
+	agg := &telemetry.Interface{Name: ygot.String(tc.aggID)}
+	tc.configAggregateDUT(agg, &dutIPs)
+	aggPath := d.Interface(tc.aggID)
+	fptest.LogYgot(t, tc.aggID, aggPath, agg)
+	aggPath.Replace(t, agg)
+
+	for _, port := range tc.dutPorts {
+		i := &telemetry.Interface{Name: ygot.String(port.Name())}
+		tc.configMemberDUT(i, port)
+		iPath := d.Interface(port.Name())
+		fptest.LogYgot(t, port.String(), iPath, i)
+		iPath.Replace(t, i)
+	}
+}
+
+func (tc *testCase) configureATE(t *testing.T) {
+	if len(tc.atePorts) < 2 {
+		t.Fatalf("Testbed requires at least 2 ports, got: %v", tc.atePorts)
+	}
+
+	// Don't use WithLACPEnabled which is for emulated Ixia LACP.
+	agg := tc.top.AddInterface(ateIPs.Name)
+	lag := tc.top.AddLAG("lag").WithPorts(tc.atePorts...)
+	lag.LACP().WithEnabled(tc.lagType == lagTypeLACP)
+	agg.WithLAG(lag)
+
+	// Disable FEC for 100G-FR ports because Novus does not support it.
+	is100gfr := false
+	for _, p := range tc.atePorts {
+		if p.Speed() == 99 {
+			is100gfr = true
+		}
+	}
+	if is100gfr {
+		agg.Ethernet().FEC().WithEnabled(false)
+	}
+
+	agg.IPv4().
+		WithAddress(ateIPs.IPv4CIDR()).
+		WithDefaultGateway(dutIPs.IPv4)
+	agg.IPv6().
+		WithAddress(ateIPs.IPv6CIDR()).
+		WithDefaultGateway(dutIPs.IPv6)
+
+	// Fail early if the topology is bad.
+	tc.top.Push(t)
+
+	ok := false
+	for n := 3; !ok; n-- {
```

Comment:
I think the original author of this test copied the topology_test as the basis for this test, and topology_test implemented retrying because it was used for troubleshooting testbeds. Normally, tests should not need to do this.

Let's replace lines 247-261 to be just:

```
tc.top.Push(t).StartProtocols(t)
```